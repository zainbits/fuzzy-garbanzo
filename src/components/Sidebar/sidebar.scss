.sidebar {
    display: flex;
    flex-direction: column;
    background-color: var(--sidebar);
    height: 100%;
    width: 50px;
    color: var(--font-sidebar);
}

.icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 40px;
    height: 40px;
    padding: 5px;
    font-size: 30px;
    color: var(--font-sidebar);
}

.icon-active {
    @extend .icon;
    border-left: red solid;
}

.footer {
    @extend .icon;
    position: absolute;
    bottom: 0px;
}

.footer-active {
    @extend .footer;
    border-left: red solid;
}