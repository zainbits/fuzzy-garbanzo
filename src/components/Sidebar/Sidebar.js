import React from 'react'
import './sidebar.scss'
import { FaGithub, FaLinkedin, FaHome } from 'react-icons/fa'
import { AiFillSetting } from 'react-icons/ai'
import { NavLink } from 'react-router-dom'
import { setCurrentNavTab, toggleSettingsNavTab } from '../../redux/engine/engine-reducer'
import { useDispatch } from 'react-redux'

const Sidebar = () => {

    const dispatch = useDispatch()

    const iconActiveHandler = (e) => {
        return e.isActive ? "icon-active" : "icon"
    }

    const footerActiveHandler = (e) => {
        return e.isActive ? "footer-active" : "footer"
    }

    const activeHandler = (type) => {
        return type === 'icon' ? iconActiveHandler : footerActiveHandler
    }

    const clickHandler = () => {
        dispatch(setCurrentNavTab('settings'))
        dispatch(toggleSettingsNavTab(true))
    }

    return (
        <div className='sidebar'>
            <div className="sidebar_tabs">
                <NavLink to="/portfolio/home" className={activeHandler('icon')}>
                    <FaHome onClick={() => { dispatch(setCurrentNavTab('closed')) }} />
                </NavLink>
                <NavLink to="/portfolio/git" className="icon">
                    <FaGithub />
                </NavLink>
                <div className='icon'>
                    <FaLinkedin />
                </div>
            </div>
            <div className="sidebar_tools">
                <NavLink to="/portfolio/settings" className={activeHandler('icon')}>
                    <AiFillSetting onClick={clickHandler} />
                </NavLink >
            </div>
        </div>
    )
}

export default Sidebar