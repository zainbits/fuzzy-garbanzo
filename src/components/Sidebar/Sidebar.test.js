import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/rootReducer'

describe('Dashboard Component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
        <Provider store={store}>
            <BrowserRouter>
                <Sidebar />
            </BrowserRouter>
        </Provider>
    );
  
    expect(component).toMatchSnapshot();
  });
});