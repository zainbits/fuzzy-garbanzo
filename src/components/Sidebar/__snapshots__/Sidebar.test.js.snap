// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render correctly in "debug" mode 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <BrowserRouter>
    <Sidebar />
  </BrowserRouter>
</ContextProvider>
`;
